metrics-core/src/main/java/com/codahale/metrics/ExponentiallyDecayingReservoir.java

```diff
@@ -95,7 +95,7 @@ public void update(long value, long timestamp) {
         try {
             final double itemWeight = weight(timestamp - startTime);
             final WeightedSample sample = new WeightedSample(value, itemWeight);
-            final double priority = itemWeight / ThreadLocalRandom.current().nextDouble();
+            final double priority = itemWeight / ThreadLocalRandomFactory.current().nextDouble();
```

Comment:
Alternate approach is to switch to Java 7, but that would break Java 6 - compatibility, which I think it's not desirable for a maintenance release .